### PR TITLE
Adding react-query

### PR DESCRIPTION
**Pull request by amirfefer**
This adds [react-query](https://react-query.tanstack.com), it should replace the need to create and manage the global state and simplify the API querying

### CODE REVIEW

File: src/Components/SourcesDropdown/index.js [hunk 1]
```diff
@@ -1,58 +1,65 @@
...
+    setChosenSource(id)
+  }
+  const dropdownItemsMapper = () => {
+    if (isLoading)
+      return (
+        <DropdownItem>
+          <Spinner size="md" isSVG />
+        </DropdownItem>
+      );
+    if (sources?.length > 0) {
```
Comment (adiabramovitch) [inline src/Components/SourcesDropdown/index.js:48]: why is there a '?'

**Comment by adiabramovitch** [comment]
looks good! Can you fix the lining? :)

File: src/Components/SourcesDropdown/index.js [hunk 2]
```diff
@@ -1,58 +1,65 @@
...
   const [isOpen, setIsOpen] = React.useState(false);
-  const [sources, setSources] = React.useState([]);
+  const [chosenSource, setChosenSource] = React.useState();
+
+  const {
+    isLoading,
+    error,
+    data: sources,
+  } = useQuery(SOURCES_QUERY_KEY, fetchSourcesList);
+
```
Comment (adiabramovitch) [inline src/Components/SourcesDropdown/index.js:23]: interesting, how doe this work? (a reminder to talk about)

File: src/Components/SourcesDropdown/index.js (hunk 1 above)
Comment (amirfefer) [reply src/Components/SourcesDropdown/index.js:48]: represents [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) :)
if `sources` is null, then `sources.length` will produce a runtime error
the optional chaining stops chaining if theres a null

**Comment by amirfefer** [comment]
@adiabramovitch, linting is fixed :)

**Pull request merged**